### PR TITLE
Fix #1843

### LINKED ISSUE #1843: Compilation issue with newer libsdl2
### Environment

* **FreeOrion Version**: master
* **Operating System**: Gentoo Linux AMD64
* **Graphic card used**: NVIDIA GeForce GTX 750 Ti
* **Fetched as** 
  * Compiled from source

### Description
After upgrading to libsdl2 >= 2.0.6 I encountered same bug https://bugs.gentoo.org/634370
```
[ 16%] Building CXX object GG/CMakeFiles/GiGiSDL.dir/src/SDL/SDLGUI.cpp.o
cd /mnt/new/o01eg/build-freeorion-master-dedicated/GG && /usr/bin/c++  -DBOOST_ALL_DYN_LINK -DBOOST_ALL_NO_LINK -DBOOST_LOG_DYN_LINK -DENABLE_BINRELOC -DFREEORION_LINUX -DGiGiSDL_EXPORTS -I/mnt/another/srcs/GIT/freeorion/GG -I/mnt/new/o01eg/build-freeorion-master-dedicated/GG  -O2 -pipe -fomit-frame-pointer -march=broadwell -mmmx -msse -msse2 -msse3 -mssse3 -mcx16 -msahf -mmovbe -maes -mpclmul -mpopcnt -mabm -mfma -mbmi -mbmi2 -mavx -mavx2 -msse4.2 -msse4.1 -mlzcnt -mrtm -mhle -mrdrnd -mf16c -mfsgsbase -mrdseed -mprfchw -madx -mfxsr -mxsave -mxsaveopt --param l1-cache-size=32 --param l1-cache-line-size=64 --param l2-cache-size=8192 -mtune=intel -O3 -DNDEBUG -fPIC   -Wall -std=c++11 -o CMakeFiles/GiGiSDL.dir/src/SDL/SDLGUI.cpp.o -c /mnt/another/srcs/GIT/freeorion/GG/src/SDL/SDLGUI.cpp
In file included from /usr/include/SDL2/SDL_config.h:10:0,
                 from /usr/include/SDL2/SDL_stdinc.h:31,
                 from /usr/include/SDL2/SDL_main.h:25,
                 from /usr/include/SDL2/SDL.h:32,
                 from /mnt/another/srcs/GIT/freeorion/GG/GG/SDL/SDLGUI.h:34,
                 from /mnt/another/srcs/GIT/freeorion/GG/src/SDL/SDLGUI.cpp:25:
/usr/include/x86_64-pc-linux-gnu/SDL2/SDL_config.h:33:26: fatal error: SDL_platform.h: No such file or directory
compilation terminated.
make[2]: *** [GG/CMakeFiles/GiGiSDL.dir/build.make:63: GG/CMakeFiles/GiGiSDL.dir/src/SDL/SDLGUI.cpp.o] Error 1
make[2]: Leaving directory '/mnt/new/o01eg/build-freeorion-master-dedicated'
make[1]: *** [CMakeFiles/Makefile2:389: GG/CMakeFiles/GiGiSDL.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

When I manually add -I/usr/include/SDL2 to path it successully compiles. 
This value already set in `SDL_INCLUDE_DIR:PATH=/usr/include/SDL2` but it isn't used.

### Expected Result
Successfull compilation on new libsdl2 version.

### Steps to reproduce
* Upgrade libsdl2 to version >= 2.0.6
* Recompile freeorion

### PR DESCRIPTION
**Pull request by o01eg**

### CODE REVIEW

**Comment by geoffthemedio** [comment]
Hopefully this doesn't break support for previous SDL versions?

**Comment by o01eg** [comment]
It adds `/usr/include/SDL2/` to include path so only issue could be collision between files but it looks like travis successfully bult it.

File: CMakeLists.txt [hunk 1]
```diff
@@ -495,7 +495,7 @@ if(NOT BUILD_HEADLESS)
             ${Boost_INCLUDE_DIRS}
             ${OPENGL_INCLUDE_DIR}
             ${GLEW_INCLUDE_DIRS}
-            ${SDL_INCLUDE_DIRS}
```
Comment (geoffthemedio) [inline CMakeLists.txt:498]: this adds `SDL_INCLUDE_DIR` but removes `SDL_INCLUDE_DIRS`. Was the latter doing anything useful, then?

Comment (o01eg) [reply CMakeLists.txt:498]: Looks like `SDL_INCLUDE_DIRS` was empty.

**Pull request merged**